src/main/Security/EncryptionController.java

```diff
@@ -0,0 +1,154 @@
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.config.TinkConfig;
+import com.google.crypto.tink.integration.gcpkms.GcpKmsClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import org.bson.Document;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.util.Objects;
+
+import static com.mongodb.client.model.Filters.eq;
+
+public class EncryptionController {
+  Logger logger;
+  MongoDatabase db;
+  Aead aead;
+
+  public static final String masterKeyUri = Objects.requireNonNull(System.getenv("MASTERKEYURI"));
+
+  public static final String credentials =
+      Objects.requireNonNull(System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));
+
+  public EncryptionController(MongoDatabase db) throws GeneralSecurityException, IOException {
+    this.db = db;
+    LogFactory l = new LogFactory();
+    logger = l.createLogger("EncryptionController");
+    generateAead();
+  }
+
+  // Generates an AEAD Object through Google Tink for encryption and decryption
+  public void generateAead() throws GeneralSecurityException, IOException {
+    TinkConfig.register();
+
+    GoogleCredentials.generateCredentials();
+
+    MongoCollection<Document> keyHandles = db.getCollection("keys", Document.class);
+    Document keyDocument = keyHandles.find(eq("keyType", "encryption")).first();
+
+    assert keyDocument != null;
+    keyDocument.remove("_id");
+    keyDocument.remove("keyType");
+
+    JSONObject keyJson = new JSONObject(keyDocument);
+
+    KeysetHandle keysetHandle =
+        KeysetHandle.read(
+            JsonKeysetReader.withJsonObject(keyJson),
+            new GcpKmsClient().withCredentials(credentials).getAead(masterKeyUri));
+
+    GoogleCredentials.deleteCredentials();
+
+    aead = keysetHandle.getPrimitive(Aead.class);
+  }
+
+  public byte[] getEncrypted(byte[] data, byte[] aad) throws GeneralSecurityException {
+    try {
+      byte[] ciphertext = aead.encrypt(data, aad);
+
+      return ciphertext;
+
+    } catch (GeneralSecurityException e) {
+      logger.error("General Security Exception thrown, encrpytion unsuccessful");
+      throw e;
+    }
+  }
+
+  public byte[] getDecrypted(byte[] ciphertext, byte[] aad) throws GeneralSecurityException {
+    try {
+      byte[] decrypted = aead.decrypt(ciphertext, aad);
+
+      return decrypted;
+    } catch (GeneralSecurityException e) {
+      logger.error("Decryption Unsuccessful, double check aead");
+      throw e;
+    }
+  }
+
+  public String encryptString(String inputString, String username) throws GeneralSecurityException {
+    //    byte[] stringBytes = inputString.getBytes(StandardCharsets.ISO_8859_1);
+    //    byte[] aad = username.getBytes();
+    //    String encryptedString =
+    //        new String(getEncrypted(stringBytes, aad), StandardCharsets.ISO_8859_1);
```

Comment:
StandardCharsets.ISO_8859_1 comes up often. Let's set it as a static variable at the top of the file and use it throughout, instead of referencing it explicitly everywhere.

Reply:
Done